app/src/main/java/com/hedvig/app/feature/keygear/KeyGearValuationActivity.kt

```diff
@@ -111,30 +112,33 @@ class KeyGearValuationActivity : BaseActivity(R.layout.activity_key_gear_valuati
             finishedUploading?.let {
                 if (finishedUploading) {
                     item?.let {
-                        startActivity(
-                            KeyGearValuationInfoActivity.newInstance(
-                                this,
-                                item.fragments.keyGearItemFragment.category,
-                                ValuationData.from(
-                                    item.fragments.keyGearItemFragment.purchasePrice!!.amount,
-                                    valuationType(item),
-                                    (item.fragments.keyGearItemFragment.valuation as KeyGearItemFragment.AsKeyGearItemValuationFixed).ratio,
-                                    (item.fragments.keyGearItemFragment.valuation as KeyGearItemFragment.AsKeyGearItemValuationFixed).valuation.amount
+                        val type = valuationType(item)
+                        if (type != null) {
+                            startActivity(
+                                KeyGearValuationInfoActivity.newInstance(
+                                    this,
+                                    item.fragments.keyGearItemFragment.category,
+                                    ValuationData.from(
+                                        item.fragments.keyGearItemFragment.purchasePrice!!.amount,
+                                        type,
+                                        (item.fragments.keyGearItemFragment.valuation as KeyGearItemFragment.AsKeyGearItemValuationFixed).ratio,
+                                        (item.fragments.keyGearItemFragment.valuation as KeyGearItemFragment.AsKeyGearItemValuationFixed).valuation.amount
+                                    )
                                 )
                             )
-                        )
-                        finish()
+                            finish()
+                        }
                     }
                 }
             }
         }
     }
 
-    private fun valuationType(item: KeyGearItemQuery.KeyGearItem): ValuationType {
-        return if (item.fragments.keyGearItemFragment.valuation is KeyGearItemFragment.AsKeyGearItemValuationFixed) {
-            ValuationType.FIXED
-        } else {
-            ValuationType.MARKET_PRICE
+    private fun valuationType(item: KeyGearItemQuery.KeyGearItem): ValuationType? {
+        return when {
```

Comment:
```suggestion
        return when (item.fragments.keyGearItemFragment.valuation) {
```
This will allow you to not repeat yourself on the following rows in the when-statement 🙂